### PR TITLE
Search and create sub groups

### PR DESCRIPTION
**Pull request by edewit**
## Motivation

Adds the ability to create groups within groups and a "global" search as defined in [marvel](https://marvelapp.com/prototype/228971ig/screen/72210499)

## Verification Steps

Go to the groups section create a group and then click on the group name.
Now you can create another group that will be the child of that pref group.
Click the action dropdown and go to search to search for groups

## Checklist:

- [x] Code has been tested locally by PR requester
- [x] User-visible strings are using the react-i18next framework (useTranslation)
- [x] Help has been implemented
- [x] axe report has been run and resulting a11y issues have been resolved
- [x] Unit tests have been created/updated
- [x] Formatting has been performed via prettier/eslint
- [x] Type checking has been performed via 'yarn check-types'

### CODE REVIEW

File: tests/cypress/support/commands.ts [hunk 1]
```diff
@@ -23,3 +23,7 @@
 //
 // -- This will overwrite an existing command --
 // Cypress.Commands.overwrite("visit", (originalFn, url, options) => { ... })
+
+Cypress.Commands.add("getBy", (selector, ...args) => {
+  return cy.get(`[data-testid=${selector}]`, ...args);
```
Comment (mfrances17) [inline tests/cypress/support/commands.ts:28]: The question is... does this save enough time to justify obscuring what cypress is doing? I'm not sure. But code looks fine. Was wondering if 'getBy' might be a little vague - maybe cy.getId or cy.getById?

File: tests/cypress/support/pages/ViewHeaderPage.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/// <reference types=".." />
+
+export default class ListingPage {
+  private actionMenu = "viewheader-action";
+
+  clickAction(action: string) {
+    cy.get("[data-cy=action-dropdown]").click().getBy(action).click();
```
Comment (mfrances17) [inline tests/cypress/support/pages/ViewHeaderPage.ts]: why use getBy for one but not the other? also, should be data-testid

**Review by mfrances17 — COMMENTED** [review]
Couple comments just on the testing aspect... will revisit to review actual search and create sub groups code and test it.

File: tests/cypress/support/pages/ViewHeaderPage.ts (hunk 2 above)
Comment (edewit) [reply tests/cypress/support/pages/ViewHeaderPage.ts]: didn't use `getBy` as it's not using `data-testid` it was already there, but I'll rename

**Review by jenny-s51 — COMMENTED** [review]
Subgroup functionality looks good @edewit ! 

From the mock it looks like when a group is clicked from the initial groups list, the subgroups should render in a separate "Child Groups" tab - it's fine if you want to address this in another PR.

Also, the empty state for subgroups is the same as the empty state for groups currently, which contains the text "no groups in this realm". It might be a good idea to create a separate empty state message for subgroups, e.g. "you have not created any child groups".

**Comment by edewit** [comment]
@jenny-s51 I'll create the tabs in a separate PR, but changed the empty text for a empty sub group

**Pull request merged**